server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -0,0 +1,97 @@
+import repository.SearchParameters;
+import repository.UserRepository;
+import services.DateConverter;
+import services.applicant.exception.ApplicantNotFoundException;
+
+public class TrustedIntermediaryService {
+
+  private final DateConverter dateConverter;
+  private final UserRepository userRepository;
+
+  @Inject
+  public TrustedIntermediaryService(DateConverter dateConverter, UserRepository userRepository) {
+    this.dateConverter = Preconditions.checkNotNull(dateConverter);
+    this.userRepository = Preconditions.checkNotNull(userRepository);
+  }
+
+  public ImmutableList<Account> retriveSearchResult(
+      SearchParameters searchParameters,
+      Optional<TrustedIntermediaryGroup> trustedIntermediaryGroup) {
+    return filterAccountsBySearchParams(searchParameters, trustedIntermediaryGroup.get());
+  }
+
+  public ImmutableList<Account> filterAccountsBySearchParams(
+      SearchParameters searchParameters, TrustedIntermediaryGroup tiGroup) {
+    ImmutableList<Account> allAccounts = tiGroup.getManagedAccounts();
+    allAccounts =
+        allAccounts.stream()
+            .filter(
+                account -> {
+                  if (searchParameters.search().isPresent()) {
+                    return account
+                        .getApplicantName()
+                        .toLowerCase(Locale.ROOT)
+                        .contains(searchParameters.search().get().toLowerCase(Locale.ROOT));
+                  }
+                  if (searchParameters.searchDate().isPresent()
+                      && !searchParameters.searchDate().isEmpty()
+                      && account.getApplicantDateOfBirth().isPresent()) {
+                    LocalDate searchDate =
+                        dateConverter.parseStringtoLocalDate(searchParameters.searchDate().get());
+                    return account.getApplicantDateOfBirth().get().equals(searchDate);
+                  }
+                  return false;
+                })
+            .collect(ImmutableList.toImmutableList());
+    return allAccounts;
+  }
+
+  public void checkFormForDobUpdate(Form<UpdateApplicantDob> form, Long applicantId)
+      throws FormHasErrorException, ApplicantNotFoundException, DateOfBirthNotInPastException,
+          IncorrectDateFormatException, MissingDateOfBirthException {
+    if (form.hasErrors()) {
+      StringBuilder errorMesssage = new StringBuilder();
+      form.errors().stream()
+          .forEach(validationError -> errorMesssage.append(validationError.message()));
+      throw new FormHasErrorException(errorMesssage.toString());
+    }
+    if (Strings.isNullOrEmpty(form.get().getDob())) {
+      throw new MissingDateOfBirthException("Please Enter a valid Date of Birth");
+    }
+    Optional<LocalDate> dobDate;
+    try {
+      dobDate = Optional.ofNullable(dateConverter.parseStringtoLocalDate(form.get().getDob()));
+    } catch (DateTimeParseException e) {
+      throw e;
+    }
+    if (isDobInPast(dobDate.get())) {
```

Comment:
Blocking: We discussed this offline (and it likely hasn't been incorporated yet). This needs a "!", otherwise it'll return an error for valid dates.